lib/Types.ts

```diff
@@ -0,0 +1,18 @@
+export type CreateRoom = { size: number }; // data sent by frontend on room creation information
+export type ConnectRoom = { roomid: string }; // data sent by frontend to connect to specific room
+export type UserDisplayMap = { [userid: string]: UserDisplay }; // Map from userid to display name
+export type UserDisplay = {
+    // user information
+    userid: string;
+    displayName: string;
+    color: string;
+};
+
+export type UserRoomMap = { [userid: string]: string }; // Map from userid to roomid (needed to handle disconnects)
+export type RoomMap = { [roomid: string]: Room }; // All open rooms and relevant information
```

Comment:
Does the `uuid` have types by any chance? Can we restrict `userid` and `roomid` to type `UUID`?

Reply:
`userid` and `roomid` values are actually `UUID`s already, but I wasn't sure how to make the type `UUID` instead of `string`.

Reply:
Jk, I don't know if the `userid` is a `UUID` because it is generated by Socket.io.